lib/kitchen/driver/ec2.rb

```diff
@@ -422,6 +423,53 @@ def submit_server
         ec2.create_instance(instance_data)
       end
 
+      def config
+        return super unless @config
+        @config
+      end
+
+      # Take one config and expand to multiple configs
+      def expand_config(conf, key)
+        configs = []
+
+        if conf[key] && conf[key].kind_of?(Array)
+          values = conf[key]
+          values.each do |value|
+            new_config = config.clone
```

Comment:
This needs to be changed from config.clone to conf.clone in order to support more than one array of parameters to be expanded. We for example used both subnet & instance type together in order to have a better chance at actually getting a spot request successfully fulfilled.
Otherwise you clone the original config, which then has the key it isn't currently expanding as an array.

Reply:
Thank you for the review, I have updated this part and making a PR upstream.